#! /bin/sh -e
## rcon_tool_debianization.dpatch by Andres Mejia <mcitadel@gmail.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Patch to modify Tools/LinuxScripts/rcon script for use in a Debian based distro.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

--- Tools/LinuxScripts/rcon~	2007-05-14 23:37:17.000000000 -0400
+++ Tools/LinuxScripts/rcon	2007-05-14 23:37:17.000000000 -0400
@@ -22,6 +22,10 @@
     Tony Jackson can be contacted at tonyj@cooldark.com
 	
 =end
+=begin
+	This script has been patched from its original source for distribution
+	with Debian systems.
+=end
 
 require 'socket'
 
@@ -103,10 +107,10 @@
 	puts 'RCon utility for Alien Arena servers'
 	puts ''
 	puts '  Usage:'
-	puts '  rcon <server:port> password <commandstring>'
+	puts '  alienarena2007-server -r <server:port> password <commandstring>'
 	puts '  Optional arguments:'
 	puts ''
-	puts '  --help  This message'
+	puts '  alienarena2007-server -r --help  This message'
 	puts ''
 	exit
 end
